lib/commands/screenshots.js

```diff
@@ -41,52 +41,62 @@ async function getScreenshotWithIdevicelib (udid, isLandscape) {
   }
 }
 
-async function isIdevicescreenshotAvailable () {
-  return !!(await fs.which('idevicescreenshot'));
+async function verifyIdeviceScreenshotAvailable () {
+  try {
+    await fs.which('idevicescreenshot');
+  } catch (err) {
+    throw new Error(`No 'idevicescreenshot' program found. To use, install ` +
+                    `using 'brew install --HEAD libimobiledevice'`);
+  }
 }
 
 commands.getScreenshot = async function () {
+  const getScreenshotFromIDS = async () => {
+    log.debug(`Taking screenshot with 'idevicescreenshot'`);
+    await verifyIdeviceScreenshotAvailable();
+    const orientation = await this.proxyCommand('/orientation', 'GET');
+    return await getScreenshotWithIdevicelib(this.opts.udid, orientation === 'LANDSCAPE');
+  };
   const getScreenshotFromWDA = async () => {
+    log.debug(`Taking screenshot with WDA`);
     const data = await this.proxyCommand('/screenshot', 'GET');
     if (!_.isString(data)) {
       throw new Error(`Unable to take screenshot. WDA returned '${JSON.stringify(data)}'`);
     }
     return data;
   };
 
+  const useIdeviceScreenshot = _.lowerCase(this.opts.realDeviceScreenshotter) === 'idevicescreenshot';
+  if (useIdeviceScreenshot) {
+    return await getScreenshotFromIDS();
```

Comment:
So, we don't even try to get the screenshot with WDA if this fails and the capability is set, right?

Reply:
correct. In the cases in question, WDA crashes and nothing further can be done, in these cases.

Reply:
it would be interesting to figure out when exactly it crashes and where.

Reply:
perhaps something can still be done there, at least not to crash it totally

Reply:
I can't reproduce, and don't have access to the app that is causing the problems.